### PR TITLE
Bongoborgo v2 (Full 'borg sprite rework, this time with less bloat!)

### PR DESCRIPTION
**Pull request by Chada-1**
Continuation of this PR: https://github.com/Aurorastation/Aurora.3/pull/8169
https://forums.aurorastation.org/topic/13847-the-full-borgo-sprite-rework/ <= Feedback thread here.

* Remove all of the old legacy Humanoid sprites as available sprite options.
* Re-sprite and replace the old legacy humanoid sprites with HD-MAD, with a version available for Syndicate too. (New humanoid sprite line, courtesy of Quill#1995 on Discord.) https://i.imgur.com/55LYZEp.png <= See here for a picture of the HD-MAD sprite line.
* Entirely removed Spiderborgs from the available sprite options and replaced them with the 'Unbranded Android' (The exoskeleton with a visible posibrain and a v. intricate electricity animation)
* Created eye overlays for all of my old treadmate/landmate sprite recolors and streamlined their sprite names to be consistent among every variant including the legacy ones.
* Created a new Treadhead sprite and eye overlays and recolor it for every department. You can see an example of one here: There are variants for every department and this is the medical one, showcasing the coil and screen losing power. https://i.imgur.com/KC4wiJe.gifv <= See here.
* Create new toiletbot sprites and eye overlays recolor them for every department. They'll be known as 'Toileto-Tron's https://i.imgur.com/MMJGuFX.gifv <= See here for an example of one.
* Implemented a new 'borg eye overlay system that allows locking them to intent, help or harm. I've decided to go with green for help and blue for harm, and I'm in the process of migrating all of my old sprites over, but they're p. much done already. I migrated all of our custom 'borgs to this system (Using two help overlays) so they shouldn't experience any difference at all in how the intent functions, it's the same for them.
* Created the new SD-MAD hoverbot series. They're be like HD-MAD but with lighter equipment and they're lifted by propulsion, so they float ALA the current Drones, they have no legs. https://i.imgur.com/SRaqRse.gifv <= See here for an example of one, we have one for every department.
* Created the new off-brand MAD hoverbot series. See above. These will have no equipment and be p. much just an endo-skeleton on the hoverbase. https://i.imgur.com/pUA6g9l.gifv <= See here, this is the sprite and animation
* Removed the 'Heavy' sprite series, it's redundant and has no neat features at all, I doubt it'll be missed, I'm sorry if it is.
* Addition of the new Repurposed Icarus Drone sprite series, it's part of the overall goal of trying to interconnect the Synths together, This replaces the current Drone 'borg sprite and it looks v. nice I think. https://i.imgur.com/ePfwhEk.gifv <= See here for the Engineering version, but it has variants for every department.
* Recolored the Medical 'Droid sprite for every department, overlay it, and call it 'Hephaestus Android', as the final branded 'borg sprite we'll have (unless people add more) https://i.imgur.com/hCkoQMs.gifv <= See here for examples, it's available in every department.
* Addition of the old style floating Drone sprite (Now titled 'Tau-Ceti-Drone'), can be seen here. https://i.imgur.com/tRgyzHL.gifv <=

### CODE REVIEW

**Comment by Chada-1** [comment]
!review

**Comment by alsoandanswer** [comment]
I'm loving the HD-MAD bots - they're beautiful.

For the Treadheads - maybe making their heads use the IPC heads would be better? They serve basically the same purpose - TV heads, and we could probably reuse like the screens for IPCs for INDIVIDUALITY.

SD-MADs look pretty OK.

Shouldn't red be harm intent for the screen? Makes more sense.

I guess for Emag/Malf you could add some glitchiness to it like static.

For the name of Toileto-tron, I think it's too on-the-nose. Something like "T0-1L3T Chassis" would still be "teehee funne toilet" but more subtle.

For Hephaestus Android and Tau-Ceti-Droid, they're quite a mouthful.
I think shortening them to Heph-Droids (hey that rhymed) and TC-Drones would be easier to remember and type out in future.

**Comment by Chada-1** [comment]
Red can't be harm intent for the new intent because it's too charged @alsoandanswer The feature wasn't meant to make 'borgs be seen as aggressive so much as give a subtle clue that they _Could_ be. If we go for red instead of blue, anyone seeing the 'borg in harm intent will instantly assume it's malicious and a danger (And likely end up getting 'borgs into a lot of trouble) when really they could be in harm intent for a lot of other reasons including just so people don't walk through them. As for renaming the 'borgs like you said, I'm fully willing to adjust their names.

File: code/modules/mob/living/silicon/robot/robot.dm [hunk 1]
```diff
@@ -826,7 +826,10 @@
 	cut_overlays()
 
 	if(stat == CONSCIOUS)
-		add_overlay("eyes-[module_sprites[icontype]]")
+		if(a_intent == "help")
```
Comment (NonQueueingMatt) [inline code/modules/mob/living/silicon/robot/robot.dm]:
```suggestion
		if(a_intent == I_HELP)
```

Comment (Chada-1) [reply code/modules/mob/living/silicon/robot/robot.dm]: I was amazed the first one worked (but it did) and kinda had my suspicions there was a better way to check for intent, I'll see if anything changed mechanically or if it was just a code changed laterish.

Comment (Chada-1) [reply code/modules/mob/living/silicon/robot/robot.dm]: I've tested it just to see if there were any differences and it works p. much identical

Comment (mikomyazaki) [reply code/modules/mob/living/silicon/robot/robot.dm]: `I_HELP` is just a define that means "help", so it will always work identically.

**Comment by Chada-1** [comment]
Wezzy there are a lot of your complaints that can't be solved and still uphold the purpose of the PR, and also some with some serious technical concerns.

On the rowtree sprites, this is one of those really _really_ iffy issues, while I would delete them 100% if it were up to me and clear them from the DMI, there's always the chance that someone will reuse them or an admin event might require them (I DON'T KNOW WHY THO), this is an issue is shared by a lot of the other ones too, and it makes me wary to remove sprites entirely from the DMI. (Yet I have deleted afew, mostly the ones with the red crosses, or I recolored them completely.)

On the treadhead, keep in mind there's the landmate, treadmate, (All of these were recolored and re-overlayed in this rework too) and then the treadhead. All related in theme kinda. The treadhead has special animations for its monitor head that you should see ingame, it looks a lot better there than it does in just the DMI, off-powered mode. If you see a way to make the head fit better I'd love to do that but I think it's an overall much more improved sprite vs the old Treadhead (You can see it in the mining section of the DMI) and fits the theme a lot better. So we still have landmate, treadmate, and the old treadheads, but I kept the old treadheads for similar reasons as I didn't delete the rowtree.

On the Repo Drones, this comes down in essence to the goal of the rework. We currently have a problem where our Synths do not share a theme or elements, they're just all over the place. The 'borg sprites are the biggest offender of this. The _goal of the rework_ is to make the 'borg sprites share elements with all of the other Synths by and large but still remain visually distinguishable and unique, the Repo Drones link the 'borgs to the other Synths in use by NanoTrasen (The Icarus Drones), the cavern dweller analogy is true, but would it really be bad to reuse the cavern dwellers in xenobiology if it were redesigned? Xenobiology has every reason to want to research them. 👀 

In the same simile the Repo Drones serve a similar purpose of tying the 'borgs to the Lore via NT re-using Icarus' Drone chassis. I'd love to improve the sprite but I have no clue how, but I'm v. convinced it's a compelling addition to the list just on that reasoning. It makes the Icarus Drones not exist in a vacuum, they're not _just_ a random event, NT actually uses them. Or in your comparison, the Cavern Dwellers aren't just a random mob, they're an actual alien lifeform that's supposed to be researched, they're actually _there_.

Before making any final judgment of the Repo Drones I really _really_ suggest you look at the animations ingame and how it looks with the overlay, I made it look p. nice even if the colors still need improvements.

The Humanoid sprites I would remove but it has a similar problem as the rowtree/old treadheads. I'm not sure when to delete and when to keep. 

The 'Heavy' sprite line was removed mostly because it doesn't fit the theme of the rest of the sprites, but with changes it could be brought back at a later time. They also have next to no animations and their overlays are p. poor so they'd need quite a bit of work.

On compiling a list of sprites, I have one on my thread [Here](https://forums.aurorastation.org/topic/13847-the-full-borgo-sprite-rework/) But it's missing afew elements, compiling a list of the current sprites not in this commit is possible but also would kinda be actual torture since it's over 12 sprite lines, and not counting re-occurring recolors.

The off-brand spizzytron works into the goal of tying the Synths together by creating a lower-grade version of the SD-MAD, it's meant to be like the other but worse in equipment, it's p. much a ragtag version of it, a worn down model. It has better animations just on the basis of having a monitor head which is a whole lot easier to animate. @alsoandanswer

**Comment by Chada-1** [comment]
@alsoandanswer The many duplicate sprites (And many of the unused ones, including the spiderborg ones) have been removed but I still would appreciate some ideas on how to improve the Treadhead (Preferably without completely having to reanimate its head, but I would if I had to, it's just a lot of work)
Then also some ideas on how to improve the Repo-Drone (Check it out ingame with the overlay and float, it looks a lot better that way.)

I purposely left the heavies/bloodhounds and a couple others in just incase I decided to update them later or someone else wanted to.

**Comment by Chada-1** [comment]
Alright @alsoandanswer Both the Heavy and Spiderborgs were re-added on your request.

**Pull request merged**